### PR TITLE
{tools}[intel/2018a] CDO v1.9.5, YAXT v0.6.0, ecCodes v2.8.2 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
{tools}[intel/2018a] CDO v1.9.5 + deps

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2586.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/695fa7b0aa61a255a9a398150d8a62c5 for a full test report.

**Comment by boegel** [comment]
I'm using `JasPer` v1.900.1 as dependency here for compatibility with `NCL` in #6698 (where JasPer 1.x is a strict requirement)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
gligar02.gligar.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/afe85cc54b6ba0723fea75667e06be8e for a full test report.

File: easybuild/easyconfigs/c/CDO/CDO-1.9.5-intel-2018a.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+checksums = ['48ed65cc5b436753c8e7f9eadd8aa97376698ce230ceafed2a4350a5b1a27148']
+
+dependencies = [
+    ('HDF5', '1.10.1'),
+    ('netCDF', '4.6.0'),
+    ('YAXT', '0.6.0'),
+    ('ecCodes', '2.8.2', '-JasPer-1.900.1'),
+]
+
+configopts = "--with-hdf5=$EBROOTHDF5 --with-netcdf=$EBROOTNETCDF --with-eccodes=$EBROOTECCODES"
```
Comment (verdurin) [inline easybuild/easyconfigs/c/CDO/CDO-1.9.5-intel-2018a.eb]: @boegel Any reason why the `grib_api` support has been removed?

Comment (boegel) [reply easybuild/easyconfigs/c/CDO/CDO-1.9.5-intel-2018a.eb]: @verdurin `ecCodes` replaces `grib_api`, see https://confluence.ecmwf.int/display/GRIB/Home

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 2 out of 4 (3 easyconfigs in this PR)
rescomp2 - Linux centos 6.8, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.7.11
See https://gist.github.com/506198242d068438a7564abf4991542f for a full test report.

**Comment by verdurin** [comment]
@boegel build failure:
```
gcc -pthread -shared -Wl,-rpath=/apps/well/python/2.7.11/lib -L/mgmt/modules/eb/software/icc/2018.1.163-GCC-6.4.0-2.28/lib/intel64 -L/mgmt/modules/eb/software/imkl/2018.1.163-iimpi-2018a/lib -L/mgmt/modules/eb/software/imkl/2018.1.163-iimpi-2018a/mkl/lib/intel64 -L/mgmt/modules/eb/software/imkl/2018.1.163-iimpi-2018a/lib -L/well/rescomp/eb/software/netCDF/4.6.0-intel-2018a/lib64 -L/well/rescomp/eb/software/JasPer/1.900.1-GCCcore-6.4.0/lib -O2 -xHost -ftz -fp-speculation=safe -fp-model source -I/mgmt/modules/eb/software/imkl/2018.1.163-iimpi-2018a/mkl/include -I/well/rescomp/eb/software/netCDF/4.6.0-intel-2018a/include -I/well/rescomp/eb/software/JasPer/1.900.1-GCCcore-6.4.0/include build/temp.linux-x86_64-2.7/dev/shm/ecCodes/2.8.2/intel-2018a-JasPer-1.900.1/eccodes-2.8.2-Source/python/swig_wrap_numpy.o build/temp.linux-x86_64-2.7/dev/shm/ecCodes/2.8.2/intel-2018a-JasPer-1.900.1/eccodes-2.8.2-Source/python/grib_interface.o -L/dev/shm/ecCodes/2.8.2/intel-2018a-JasPer-1.900.1/easybuild_obj/lib -L/apps/well/python/2.7.11/lib -Wl,-R/dev/shm/ecCodes/2.8.2/intel-2018a-JasPer-1.900.1/easybuild_obj/lib -leccodes -lpython2.7 -o /dev/shm/ecCodes/2.8.2/intel-2018a-JasPer-1.900.1/easybuild_obj/python/gribapi/_gribapi_swig.so
gcc: error: source: No such file or directory
gcc: error: unrecognized command line option ‘-ftz’; did you mean ‘-flto’?
gcc: error: unrecognized command line option ‘-fp-speculation=safe’
gcc: error: unrecognized command line option ‘-fp-model’
error: command 'gcc' failed with exit status 1
make[2]: *** [python/gribapi/_gribapi_swig.so] Error 1
make[2]: Leaving directory `/dev/shm/ecCodes/2.8.2/intel-2018a-JasPer-1.900.1/easybuild_obj'
make[1]: *** [python/CMakeFiles/eccodes_build_swig_wrapper.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by boegel** [comment]
@verdurin Hmm, that's pretty strange, since I'm not seeing that problem at all.

There's no trace of `gribapi` in my installation log though, so it seems like something extra is being installed in your case, where `gcc` is hardcoded, which makes it go down...

I see this in the output during the `configure` step, I suspect you have something different?

```
-- Found PythonInterp: /usr/bin/python (found version "2.7.5")
-- Found Python: TRUE
-- Found package Python required for feature PYTHON
-- Could NOT find package NumPy required for feature PYTHON -- Provide NumPy location with -DNUMPY_PATH=/...
-- Feature PYTHON was not enabled (also not requested) -- following required packages weren't found: NumPy
```

Can you share a full debug log so I can compare?

**Comment by boegel** [comment]
@verdurin I expect that the problem is fixed by disabling the Python bindings for `ecCodes`, see changes in 94837e5eb .

Please retest?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2565.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/2e4cb48d2f49b27f164309c7c7469418 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 2 out of 2 (3 easyconfigs in this PR)
rescomp2 - Linux centos 6.8, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.7.11
See https://gist.github.com/4757cef68a84ccc67eef5e76e981b5fc for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3109.skitty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/61880241a03b5b107691671c9e4e400c for a full test report.

**Comment by boegel** [comment]
marking this as WIP, since there seems to be a trivial workaround for the JasPer 2.x incompatibility, cfr. patches for `g2lib` & `g2clib` in #5851

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2627.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b58a043bb15709f3925a9c98263d9b73 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
gligar01.gligar.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/37c2eaef1cb6a6082d3bb967718ba6f1 for a full test report.

**Comment by boegel** [comment]
@verdurin Good to go now, #6698 no longer needs `JasPer` 1.x

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3108.skitty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/eaa52e652faaedfaa9d8a6623cf02da1 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (3 easyconfigs in this PR)
rescomp2 - Linux centos 6.8, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.7.11
See https://gist.github.com/f66b99f19aadaaab5eba14d69f52079d for a full test report.

**Comment by verdurin** [comment]
Going in, thanks @boegel!

**Pull request merged**